h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -519,6 +531,15 @@ public void init(boolean expensive) {
       _t0 = System.currentTimeMillis();
       if ((_parms._lambda_search || !_parms._intercept || _parms._lambda == null || _parms._lambda[0] > 0) && !_parms._HGLM)
         _parms._use_all_factor_levels = true;
+      if (_parms._family == Family.AUTO) {
+        if (_nclass == 1) {
+          _parms._family = Family.gaussian;
+        } else if (_nclass == 2) {
+          _parms._family = Family.binomial;
+        } else {
+          _parms._family = Family.multinomial;
+        }
+      }
       if (_parms._link == Link.family_default)
```

Comment:
If you move line 534 to 542 before line 442, you don't need to worry about any options for Family.AUTO, right?

Reply:
Yesyes - I just thought that when you input AUTO and incompatible link for given response and get error that AUTO requires different link for this response, it could be more clear than getting error that your link is incomatible for some specific family which you haven't input - but in the end it's the same thing, and will tell you which link is compatible so I can move it.